src/openvic-simulation/scripts/Condition.cpp

```diff
@@ -0,0 +1,659 @@
+#include "Condition.hpp"
+#include "openvic-simulation/GameManager.hpp"
+#include <vector>
+
+using namespace OpenVic;
+using namespace OpenVic::NodeTools;
+
+Condition::Condition(
+	std::string_view new_identifier, value_type_t new_value_type, scope_t new_scope,
+	scope_t new_scope_change, identifier_type_t new_key_identifier_type,
+	identifier_type_t new_value_identifier_type
+) : HasIdentifier { new_identifier }, value_type { new_value_type }, scope { new_scope },
+	scope_change { new_scope_change }, key_identifier_type { new_key_identifier_type },
+	value_identifier_type { new_value_identifier_type } {}
+
+ConditionNode::ConditionNode(
+	Condition const* new_condition, value_t&& new_value,
+	HasIdentifier const* new_condition_key_item,
+	HasIdentifier const* new_condition_value_item
+) : condition { new_condition }, condition_key_item { new_condition_key_item },
+	condition_value_item { new_condition_key_item }, value { std::move(new_value) } {}
+
+bool ConditionManager::add_condition(
+	std::string_view identifier, value_type_t value_type, scope_t scope, scope_t scope_change,
+	identifier_type_t key_identifier_type, identifier_type_t value_identifier_type
+) {
+	if (identifier.empty()) {
+		Logger::error("Invalid condition identifier - empty!");
+		return false;
+	}
+
+	if (value_type == 0) {
+		Logger::error("Condition ", identifier, " accepts no value type!");
+		return false;
+	}
+
+	if (value_type > 0b1111111) {
+		Logger::error("Condition ", identifier, " accepts an invalid value type!");
+		return false;
+	}
+
+	if (scope == 0) {
+		Logger::error("Condition ", identifier, " has no scope specified!");
+		return false;
+	}
+
+	if (scope > (scope_t::COUNTRY | scope_t::PROVINCE | scope_t::POP | scope_t::MAGIC)) {
+		Logger::error("Condition ", identifier, " has no scope specified!");
+		return false;
+	}
```

Comment:
```suggestion
	if (value_type == 0 || value_type > value_type_t::MAX_VALUE) {
		Logger::error("Condition ", identifier, " has invalid value type: ", static_cast<uint64_t>(value_type));
		return false;
	}

	if (scope == scope_t::UNSPECIFIED || scope > scope_t::MAX_SCOPE) {
		Logger::error("Condition ", identifier, " has invalid scope: ", static_cast<uint64_t>(scope));
		return false;
	}
```
- Better to use constants defined in the enums rather than hardcoded numbers or manually combining all the scopes, simplest way to define them is: `MAX_VALUE = (1 << 8) - 1` (don't worry about `1 << 8` being bigger than a `uint8_t`, the compiler works the whole constant out before making it 8-bit) and `MAX_SCOPE = (1 << 4) - 1`. If these are defined as enum values it'll be easier to update them if new values are ever added.
- The values are cast to `uint64_t`s in the error message as 8-bit values by default are treated as characters, not numbers, so it needs to be cast to at least 16-bit, and 64-bit covers any future enlargement of the enum type.